python/lsst/pipe/tasks/transformMeasurement.py

```diff
@@ -152,38 +138,85 @@ class RunTransformTask(pipeBase.CmdLineTask):
 
     @pipeBase.timeMethod
     def run(self, dataRef):
         """!Transform the source catalog referred to by dataRef.
 
-        The result is both returned and written as dataset type "transformedSrc"
-        to the provided dataRef.
+        The result is both returned and written as dataset type "transformed" + the input
+        source type to the provided dataRef.
 
-        @param[in] dataRef  Data reference for source catalog (src) &
-                            calibrated exposure (calexp).
+        @param[in] dataRef  Data reference for source catalog & calibrated exposure (calexp).
 
         @returns A BaseCatalog containing the transformed measurements.
         """
-        inputCat = dataRef.get('src')
-        wcs = dataRef.get('calexp').getWcs()
-        calib = dataRef.get('calexp').getCalib()
+        inputCat = dataRef.get(self.config.sourceType)
+        wcs = dataRef.get(self.config.calexpType).getWcs()
+        calib = dataRef.get(self.config.calexpType).getCalib()
         outputCat = self.transform.run(inputCat, wcs, calib)
-        dataRef.put(outputCat, "transformedSrc")
+        dataRef.put(outputCat, "transformed" + self.config.sourceType.capitalize())
         return outputCat
+
+
+def makeTransformCmdLineTask(sourceType, calexpType, measConfigType, outputType, wasForced):
+    """!Generate and return a specialization of RunTransformTaskBase.
+
+    RunTransformTaskBase provides a means of transforming arbitrary source types (src, forced_src,
+    deepCoadd_src, ...). However, a specialization is required for each type we wish to transform. Here, we
+    generate a specialization for the source type required.
+
+    @param[in] sourceType      Name of source type to transform (e.g src, forced_src, etc).
+    @param[in] calexpType      Name of exposure with calibration data (e.g. calexp, etc)
+    @param[in] measConfigType  Name of measurement task configuration (e.g. processCCd_config, etc)
+    @param[in] outputType      Name of output type to write (e.g. transformedSrc, etc)
+    @param[in] wasForced       Boolean indicating if source measurements were forced.
+
+    @returns A tuple of (Config, CmdLineTask) suitable for transforming sources of the given type.
+    """
+    sourceTypeField = pexConfig.Field(dtype=str, default=sourceType,
+                                      doc="Dataset type of source to transform")
+    calexpTypeField = pexConfig.Field(dtype=str, default=calexpType,
+                                      doc="Dataset type of calibration exposure")
+    measConfigField = pexConfig.Field(dtype=str, default="processCcd_config",
+                                      doc="Dataset type of measurement operation configuration")
+    outputTypeField = pexConfig.Field(dtype=str, default=outputType,
+                                      doc="Dataset type of output sources")
+    forcedField = pexConfig.Field(dtype=bool, default=wasForced, doc="Measurement operation was forced")
```

Comment:
I don't think most of these should be config fields; I think class attributes would be sufficient for most cases.

Unfortunately, that's not sufficient for all cases, because the coadd outputs are indeed dependent on their configuration - but they're dependent on configuration options that are different from the ones here (i.e. one configures only the prefix, not the whole dataset name).  And for those, I think it'd be nice to make it possible to read the config values from the original processing task, the way the measurement config is.  I'll discuss this more elsewhere, but I think this is pushing me towards preferring having a base class for transformations that is manually subclassed for each specialization, rather than having a generating function for the new tasks.